File: internal/theme/domain/theme.go
package domain

const (
	// DefaultTheme 默认主题
	DefaultTheme = "dark"
	// ThemeLight 浅色主题
	ThemeLight = "light"
	// ThemeDark 深色主题
	ThemeDark = "dark"
)

// ValidThemes 有效的主题列表
var ValidThemes = []string{ThemeLight, ThemeDark}

// ValidateTheme 验证主题值
// 如果主题无效，返回默认主题
func ValidateTheme(theme string) string {
	if theme == ThemeLight || theme == ThemeDark {
		return theme
	}
	return DefaultTheme
}

// IsValidTheme 检查主题值是否有效
func IsValidTheme(theme string) bool {
	return theme == ThemeLight || theme == ThemeDark
}

File: frontend/src/utils/theme.js
/**
 * 主题管理工具
 */

const DEFAULT_THEME = 'dark'
const VALID_THEMES = ['light', 'dark']

/**
 * 验证主题名称
 * @param {string} theme - 主题名称
 * @returns {string} 有效的主题名称
 */
function validateTheme(theme) {
  return VALID_THEMES.includes(theme) ? theme : DEFAULT_THEME
}

/**
 * 加载并应用主题
 * @param {string} theme - 主题名称 ('light' | 'dark')
 */
export function loadTheme(theme) {
  const validTheme = validateTheme(theme)
  if (theme !== validTheme) {
    console.warn(`无效的主题: ${theme}，使用默认主题 ${validTheme}`)
  }
  
  // 设置 data-theme 属性到 html 元素
  document.documentElement.setAttribute('data-theme', validTheme)
}

/**
 * 获取默认主题
 * @returns {string} 默认主题名称
 */
export function getDefaultTheme() {
  return DEFAULT_THEME
}

